hbase_sql/src/main/java/com/splicemachine/olap/OlapServerSubmitter.java

```diff
@@ -166,10 +163,13 @@ public void run() {
                         public void run() {
                             RecoverableZooKeeper zk = ZkUtils.getRecoverableZooKeeper();
                             String root = HConfiguration.getConfiguration().getSpliceRootPath();
-                            String path = root + HBaseConfiguration.OLAP_SERVER_PATH + HBaseConfiguration.OLAP_SERVER_KEEP_ALIVE_PATH + "/" + appName;
-                            byte[] payload = Bytes.toBytes(System.currentTimeMillis());
+                            String keepAlivePath = root + HBaseConfiguration.OLAP_SERVER_PATH + HBaseConfiguration.OLAP_SERVER_KEEP_ALIVE_PATH + "/" + appName;
+
+                            byte[] payload = getKeepAlivePayload();
                             try {
-                                zk.create(path, payload, ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.EPHEMERAL);
+                                // clean up node if it exists
+                                ZkUtils.safeDelete(keepAlivePath, -1);
+                                zk.create(keepAlivePath, payload, ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.EPHEMERAL);
                             } catch (KeeperException ke) {
                                 if (ke.code().equals(NODEEXISTS)) {
```

Comment:
Is this code possible after the change? Should we still ignore it?

Reply:
I removed it, good point